### PR TITLE
Fixing Testing Controller sample

### PR DESCRIPTION
**Pull request by kichalla**
@Rick-Anderson @pranavkm

### CODE REVIEW

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/project.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿{
+  "dependencies": {
```
Comment (kichalla) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/project.json]: @pranavkm Could you check if this looks fine?

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/IntegrationTests/TestFixture.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace TestingControllersSample.Tests.IntegrationTests
+{
+    public class TestFixture<TStartup> : IDisposable
```
Comment (kichalla) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/IntegrationTests/TestFixture.cs]: @pranavkm Could you check if this looks fine?

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Controllers/HomeController.cs [hunk 3]
```diff
@@ -17,16 +18,17 @@ public HomeController(IBrainstormSessionRepository sessionRepository)
...
-                {
-                    Id = s.Id,
-                    DateCreated = s.DateCreated,
-                    Name = s.Name,
-                    IdeaCount = s.Ideas.Count
-                });
+            var sessionList = await _sessionRepository.ListAsync();
+
+            var model = sessionList.Select(session => new StormSessionViewModel()
+               {
```
Comment (pranavkm) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Controllers/HomeController.cs]: de-dent this?

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Startup.cs [hunk 4]
```diff
@@ -1,56 +1,55 @@
...
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
-using System.Linq;
+using Microsoft.EntityFrameworkCore;
 using TestingControllersSample.Core.Interfaces;
 using TestingControllersSample.Core.Model;
 using TestingControllersSample.Infrastructure;
+using System.Threading.Tasks;
```
Comment (pranavkm) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/Startup.cs]: Sort usings.

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/project.json [hunk 5]
```diff
@@ -1,36 +1,51 @@
...
+      "version": "1.0.0"
+    }
   },
 
-  "commands": {
-    "web": "Microsoft.AspNet.Server.Kestrel"
+  "frameworks": {
+    "netcoreapp1.0": {
+      "imports": [
+        "dotnet5.6",
```
Comment (pranavkm) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/project.json]: Shouldn't need imports.

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/IntegrationTests/TestFixture.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        {
+        }
+
+        protected TestFixture(string solutionRelativePath)
+        {
+            var startupAssembly = typeof(TStartup).GetTypeInfo().Assembly;
+            var contentRoot = GetProjectPath(solutionRelativePath, startupAssembly);
+
+            var builder = new WebHostBuilder()
+                .UseContentRoot(contentRoot)
```
Comment (pranavkm) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/IntegrationTests/TestFixture.cs]: I couldn't quite tell - but do we do views tests in this sample? If there are only API tests, we could drop this code (and finding the sln snippet). Alternatively add a test for views (assuming it's not already there)

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/project.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    "System.Diagnostics.TraceSource": "4.0.0",
+    "System.Net.Http": "4.1.0",
+    "TestingControllersSample": "1.0.0",
+    "xunit": "2.1.0"
+  },
+
+  "frameworks": {
+    "netcoreapp1.0": {
+      "imports": [
+        "dnxcore50",
```
Comment (pranavkm) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/project.json]: Shouldn't need this one.

**Comment by pranavkm** [comment]
:watch: for the few questions.

File: aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/project.json (hunk 5 above)
Comment (kichalla) [inline aspnet/mvc/controllers/testing/sample/TestingControllersSample/src/TestingControllersSample/project.json]: Looks like I need this currently as my package restores failed on Linux for Moq related dependencies

**Comment by kichalla** [comment]
:up: :date:

**Comment by pranavkm** [comment]
:shipit:

**Comment by kichalla** [comment]
@Rick-Anderson Could you please merge this PR?

**Pull request merged**